jpc/src/lib.rs

```diff
@@ -2898,13 +2934,12 @@ impl ContiguousCodestream {
 
         // The tile-part headers are found at the beginning of each tile-part
         let tile_header = self.decode_first_tile_header(reader, no_components)?;
-        let mut marker_type: MarkerSymbol = [0; 2];
 
         // Required as the last marker segment of every tile-part header
-        reader.read_exact(&mut marker_type)?;
+        let marker_type: MarkerSymbol = MarkerSymbol::decode(reader)?;
         if marker_type != MARKER_SYMBOL_SOD {
             return Err(CodestreamError::MarkerUnexpected {
-                marker: MARKER_SYMBOL_SOD,
+                marker: marker_type,
```

Comment:
Behaviour change: previously we flagged the marker we were expecting. Now we flag the marker we found.